### ISSUE TITLE
T2.1: Remove `Committable` bound from `Block` trait

### ISSUE THREAD

**Issue by ggutoski** [EspressoSystems/HotShot#1757]
- [x] #1772 
- [x] #1778
- [x] #1788 
- [x] #1791 
- [x] #1793 
- [ ] everything else needed to finish the task

# Why?

We could make it so that `block.commit()` returns the VID commitment. But that approach has knock-on effects. Example: VID commitments are expensive to compute, but comes for free when computing VID shares. Thus, an efficient solution would entangle VID into the `Block`, which is complex and not modular.

So don't allow the user to call `block.commit()`. Drop support for `Committable` at the block level and instead ask the user to do something like `vid.commit(block.serialize())`. This approach is modular -- a block doesn't need to know anything about VID or commitments.

**Comment by ggutoski** [comment]
Lots of progress has been made on this task, and the codebase is better for it, but more work remains before we can remove `Committable` from `BlockPayload`. We recently made a new plan that makes it unnecessary to complete this task: https://github.com/EspressoSystems/HotShot/issues/1817#issuecomment-1747143528 . I still prefer that `BlockPayload` is not `Committable` but we should not devote further resources to this task.
